packages/fogbender-element/src/utils.ts

```diff
@@ -13,23 +15,35 @@ export const noopCleanup = () => {
   return new Promise<() => void>(resolve => resolve(() => {}));
 };
 
-export const renderIframe = (divRef: HTMLDivElement | undefined, headless: boolean) => {
-  const fogbender = useContext(FogbenderContext);
+export const renderIframe = (
+  divRef: HTMLDivElement | undefined,
+  headless: boolean,
+  element: ICustomElement
+) => {
+  const fogbender: Fogbender = consume(fogbenderContext, element as HTMLElement & ICustomElement);
```

Comment:
why do we need `element` here?

Reply:
Okay! So, this particular part is tricky one
![image](https://user-images.githubusercontent.com/47587853/192138333-bf8f6039-5108-4a2e-bf4c-b92e6c8f3fd0.png)

Connected callback ran when a component is mounted on the `DOM` and  initially current element is null.
![image](https://user-images.githubusercontent.com/47587853/192138389-1fcb8456-b21b-4638-abb7-b0751f38dd57.png)

In connectedCallback in finally block we are assigning current element to outer element at the end which basically is undefined so when the function is executed completely current element will be undefined.

This getCurrentElement function is giving the instance of current element
![image](https://user-images.githubusercontent.com/47587853/192138923-61554090-d344-4436-a3a6-ff6cd3ed3c83.png)


So when we are consuming context using consume function, at the time we reach consume in simple widget current element becomes undefined as connected callback is executed completely
![image](https://user-images.githubusercontent.com/47587853/192138995-e734e740-7e30-43c6-85f7-cd5340bf7845.png)

And consume is using this lookup context 
![image](https://user-images.githubusercontent.com/47587853/192139023-3c59966f-dfa2-41cd-8b73-d73871711f96.png)

So long story short is lookup context needs a element ref in order to check its parent node for context.
So that's why I have pass the element from outside

Reply:
I think was also the side effect of shadowDOM. This is also removed now.

Reply:
![image](https://user-images.githubusercontent.com/47587853/192143656-36d929d1-c95a-458d-9075-7ead553256d8.png)

If no element is passed then when we change the props on the run time from the DOM then this error come.